docs/flashing.md

```diff
@@ -13,16 +13,44 @@ coreboot's `cbmem` tool can be used to verify this. The call to
 `FspMemoryInit()` can report 20+ seconds on the first boot, and a few hundred
 milliseconds on subsequent boots.
 
+## UEFI application
+
+This is the default method for flashing firmware (using firmware-update). When
+used from this repo, it only works with systems running System76 open firmware.
+
+```
+./scripts/flash.sh <model>
+```
+
+By default the script will attempt to flash both the BIOS and the EC. Their
+respective file can be deleted to skip flashing them.
+
+If the EC is flashed, the system will immediately power off.
+
 ## Internal programmer
 
-Use this method for flashing a system already running System76 Open Firmware.
+### Availability
+
+This method is only possible when firmware is unlocked. Firmware is unlocked by
+one of two methods:
+
+- The EC feature `CONFIG_SECURITY` is unset/disabled
+- The EC is unlock for a single boot (via firmware-update or ectool)
+
+The current state can be determined using ectool:
 
 ```
-./scripts/flash.sh <model> [--without-ec]
+./ec/scripts/ectool.sh security
 ```
 
-By default the script will attempt to flash the EC. If the EC is flashed, the
-system will immediately power off.
+- `Lock`: This method can't be used
+- `Unlock`: This method can be used
+
+### flashrom
+
+```
+sudo flashrom -p internal -w build/<model>/firmware.rom
```

Comment:
When I attempt to run `sudo flashrom -p internal -w build/<model>/firmware.rom` for addw3, I reliably get a "No chipset found" warning and a failure to detect the flash device.

I've confirmed that `ectool.sh security` returns `unlock` (Though that took two boots to actually unlock - one reboot, one cold boot) 

I also built flashrom from our [fork](https://github.com/system76/flashrom/tree/main) in case the version in the repos was too old to support the addw3, but that had the same result. 

Any guidance?

Reply:
I don't think it works with ADL+, but I'm not sure if that's a problem with flashrom or coreboot/Linux (exposing the device).